src/components/Descope.tsx

```diff
@@ -114,14 +115,45 @@ const Descope = React.forwardRef<HTMLElement, DescopeProps>(
 			const ele = innerRef;
 			ele?.addEventListener('success', handleSuccess);
 			if (onError) ele?.addEventListener('error', onError);
+			if (onReady) ele?.addEventListener('ready', onReady);
 
 			return () => {
 				if (onError) ele?.removeEventListener('error', onError);
+				if (onReady) ele?.removeEventListener('ready', onReady);
 
 				ele?.removeEventListener('success', handleSuccess);
 			};
 		}, [innerRef, onError, handleSuccess]);
 
+		// Success event
+		useEffect(() => {
+			const ele = innerRef;
+			ele?.addEventListener('success', handleSuccess);
+			return () => {
+				ele?.removeEventListener('success', handleSuccess);
+			};
+		}, [innerRef, handleSuccess]);
+
+		// Error event
+		useEffect(() => {
+			const ele = innerRef;
+			if (onError) ele?.addEventListener('error', onError);
+
+			return () => {
```

Comment:
I was debating myself weather to
1. return a cleanup function that checks inside if `onError`
2. only return a cleanup function if `onError`

I think both approaches are quite equivalent, 1 is a bit more readable and simple to maintain (at least imo), while 2 has a slightly better efficiency

I chose 1 because I think the efficiencey here is very neglectable, lmk if you think otherwise